### DISCUSSION TITLE
How to use aggrid.from_pandas with different options?

### DISCUSSION THREAD

**Discussion by clj55** [zauberzeug/nicegui#947] [Q&A] [answered]
### Question

Using from_pandas method, I can directly pass the pandas dataframe to the grid. But to define additional options like `filter`,` defaultColDef`, `rowSelection` etc, I need to do `grid.options['defaultColDef']`, `grid.options['rowSelection']` one by one, Is there a way to configure all the options together like in the usual ui.aggrid()?

I tried doing
```py
grid = ui.aggrid({
        'defaultColDef':{
            'sortable':True,
            'floatingFilter':True,
            'suppressMenu':True,
        }
    }).from_pandas(to_display_df).style(grid_style)
```
But it renders a loading screen
![image](https://github.com/zauberzeug/nicegui/assets/103747880/8737154c-4473-4a68-b248-3febff788f6d)
 and my aggrid below it

**Comment by falkoschindler** [comment, marked as answer]
You can use the `update` method of Python's dictionaries to "add" another dictionary while keeping already existing key-value pairs:

```py
df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
grid = ui.aggrid.from_pandas(df)
grid.options.update({
    'defaultColDef': {
        'sortable': True,
        'floatingFilter': True,
        'suppressMenu': True,
    }
})
```

**Reply by rohitsathish** [reply]
Is there a way to do this for ```ui.table.from_pandas```?

**Reply by falkoschindler** [reply]
Have you tried?

**Reply by falkoschindler** [reply]
Or asked differently: What are you trying to do? What property do you want to change after creating the table with `from_pandas`?
